src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -170,6 +175,26 @@ protected function &getProperty(&$element, $property, $fallback = null)
     }
 
     /**
+     * set and return a property on an object or array
+     *
+     * @param mixed $element Element to manipulate
+     * @param string $property  Property to set
+     * @param mixed $value Value to set
+     */
+    protected function &setProperty(&$element, $property, $value)
```

Comment:
To avoid an extra call to getProperty, and for consistency with that method. The reference is to the newly set property, not to $value (unless $element isn't an object or an array, in which case it behaves the same as $fallback from getProperty).

Would you rather I implemented it without the reference return? The way I've done it feels tidier, to me, but quite happy to change it if you don't like that approach.